### PR TITLE
Track Chooser and LocalStorage

### PR DESCRIPTION
**Pull request by whomwah**
https://github.com/kyan/jukebox-js/issues/69

This PR adds `LocalStorage`. It's called `Settings` in the app. It's an `npm` that writes data to the FS. It's there to store useful state. For example storing the last X tracks played so you can use them without having to make an API call.

The PR also adds a lib that chooses tracks when none are left. This is like BRH but uses the Spotify recommendation API. It seems to work well for a first pass, but I plan to test it for long periods when I get back from hols.

**NOTE**
There are a couple of `istanbul ignore next`'s in here (Bad Man). I will fix these specs when I get back, but they're a bit complex and I wanted to get the code up and passing.

### CODE REVIEW

File: backend/src/lib/local-storage/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  addToUniqueArray: (key, value, limit) => {
+    let ary = Settings.getItem(key) || []
+    if (ary.constructor !== Array) {
+      throw new Error(`addToUniqueArray: ${key} is currently NOT an Array`)
+    }
+    ary.push(value)
+
+    if (limit) {
+      ary = ary.reverse().slice(0, Number(limit)).reverse()
```
Comment (atothewest) [inline backend/src/lib/local-storage/index.js]: (minor) I think `ary.splice(0, ary.length - Number(limit))` does the same thing in a single function call.

File: backend/src/lib/services/spotify/index.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+/* istanbul ignore next */
+const recommendTracks = (uris, mopidy) => {
+  const guids = uris.slice(-5).map(uri => uri.split(':').slice(-1)[0])
+
+  const options = {
+    limit: 2,
+    country: 'GB',
+    min_tempo: 140,
```
Comment (atothewest) [inline backend/src/lib/services/spotify/index.js]: Seems like quite a high min tempo 😂

File: backend/src/lib/services/spotify/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  }
+}
+
+describe('SpotifyService', () => {
+  const mockFn = jest.fn()
+
+  describe('canRecommend', () => {
+    it('should handle both cases', () => {
+      SpotifyService.canRecommend(mopidy, mockFn)
+      SpotifyService.canRecommend(mopidy, mockFn)
```
Comment (atothewest) [inline backend/src/lib/services/spotify/index.spec.js]: Does this need expectations?

File: backend/src/lib/services/spotify/index.js (hunk 2 above)
Comment (whomwah) [reply backend/src/lib/services/spotify/index.js]: 😆

File: backend/src/lib/services/spotify/index.spec.js (hunk 3 above)
Comment (whomwah) [reply backend/src/lib/services/spotify/index.spec.js]: Yes it does, doh! I need to have a think about testing in general as it seems that Enzyme is proving harder to work with as it's not keeping up with React. For instance you can't test hooks correctly currently.

**Pull request merged**